docs/community/rust-coding-guidelines/rust-secure-coding-guidelines.md

```diff
@@ -0,0 +1,253 @@
+---
+id: rust-secure-coding-guidelines
+title: Secure Coding
+---
+
+These Rust Secure Coding Guidelines are essential for anyone contributing to Aptos, reflecting our security-first approach. As Aptos is built with a primary focus on security, these guidelines, derived and adapted from ANSSI's Secure Rust Guidelines, are integral to maintaining the high standards of safety and robustness in aptos-core. Aptos contributors are encouraged to thoroughly understand and apply these principles in their work.
+
+## Development Environment
+
+### Rustup
+
+Utilize Rustup for managing Rust toolchains. However, keep in mind that, from a security perspective, Rustup performs all downloads over HTTPS, but it does not yet validate signatures of downloads. Security is shifted to [create.io](http://create.io) and GitHub repository hosting the code [[rustup]](https://anssi-fr.github.io/rust-guide/02_devenv.html#rustup).
```

Comment:
I think we should remove per-item references to the ANSSI guidelines and only link them at the end as references. We say at the top that our guidelines are based on the ANSSI ones, we don't have to match them.